packages/xstate-react/src/createProvider.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react';
+import { createContext, useContext } from 'react';
+import { useInterpret } from './useInterpret';
+import { useActor } from './useActor';
+import { useSelector } from './useSelector';
+import { ActorRefFrom, AnyStateMachine, EmittedFrom, EventFrom } from 'xstate';
+
+export function createProvider<TMachine extends AnyStateMachine>(
+  machine: TMachine
+): {
+  (props: { children: React.ReactNode }): React.ReactElement<any, any>;
+  useActor: () => [
+    EmittedFrom<ActorRefFrom<TMachine>>,
+    (event: EventFrom<TMachine>) => void
+  ];
+  useSelector: <T>(selector: (snapshot: EmittedFrom<TMachine>) => T) => T;
+  useContext: () => ActorRefFrom<TMachine>;
+} {
+  const ReactContext = createContext(null as any);
+
+  function Provider({ children }: { children: React.ReactNode }) {
+    const actor = useInterpret(machine);
+
+    return (
+      <ReactContext.Provider value={actor}>{children}</ReactContext.Provider>
```

Comment:
I think here instead of coupling with the `actor`, it should be open for the user to pass in their own dependencies here.
basically `value={actor}` => `value={value?? actor}`

Reply:
but the dependency is the actor! what other value can we pass here? 
and also, in this case, it's exactly like react context

Reply:
> 

So you're thinking of this usage?

```jsx
// another actor of same type
<SomeProvider value={anotherActor}>
  // ...
</SomeProvider>
```

And also:
```jsx
// another machine of same type
<SomeProvider value={anotherMachine}>
  // ...
</SomeProvider>
```